src/base/array_view.hpp

```diff
@@ -50,48 +48,43 @@ class ArrayView
   using const_iterator = const_pointer;
 
   ArrayView() = default;
-  constexpr ArrayView(const T* pData, const size_type size) noexcept
+  constexpr ArrayView(const_pointer pData, const size_type size) noexcept
     : mpData(pData)
     , mSize(size)
   {
   }
 
   // implicit on purpose
   template <std::size_t N>
-  constexpr ArrayView(const std::array<T, N>& array) noexcept // NOLINT
-    : mpData(array.data())
-    , mSize(static_cast<size_type>(N))
-  {
-  }
-
-  // implicit on purpose
-  template <std::size_t N>
-  constexpr ArrayView(const T (&array)[N]) noexcept // NOLINT
+  constexpr ArrayView(const value_type (&array)[N]) noexcept // NOLINT
     : mpData(array)
     , mSize(static_cast<size_type>(N))
   {
   }
 
   // implicit on purpose
-  ArrayView(const std::vector<T>& vec) noexcept // NOLINT
-    : mpData(vec.data())
-    , mSize(static_cast<size_type>(vec.size()))
+  template <typename Container>
```

Comment:
this allows use with std::array, std::vector, static_vector, even string..

Reply:
👍🏻